src/utils/array.tsx

```diff
@@ -0,0 +1,18 @@
+class ArrayClass extends Array {
```

Comment:
Same question as for the extract. Why is this a class instead of just a function?